ghost/members-api/lib/repositories/MemberRepository.js

```diff
@@ -281,7 +282,18 @@ module.exports = class MemberRepository {
             memberStatusData.status = 'comped';
         }
 
-        // Subscribe member to default newsletters
+        //checks for custom signUp forms
+        if (memberData.newsletters && memberData.newsletters.length > 0) {
+            const savedNewsletter = await this._newslettersService.browse({filter: `id:'${memberData.newsletters[0].id}'`});
+            if (savedNewsletter.length === 0) {
+                throw new errors.BadRequestError({message: tpl(messages.invalidNewsletterId, {id: memberData.newsletters[0].id})});
+            }
+            if (savedNewsletter[0].status === 'archived') {
+                memberData.newsletters = [];
```

Comment:
Doing this is important because otherwise it was still subscribing to the archived newsletter.